addons/l10n_in_pos/models/product_template.py

```diff
@@ -12,3 +14,8 @@ def _load_pos_data_fields(self, config_id):
         if self.env.company.country_id.code == 'IN':
             fields += ['l10n_in_hsn_code']
         return fields
+
+    @api.onchange('l10n_in_hsn_code')
+    def _onchange_l10n_in_hsn_code(self):
+        if self._origin.l10n_in_hsn_code and self.env['pos.session'].search_count([("company_id.account_fiscal_country_id.code", "=", "IN"), ("state", "!=", "closed")]):
+            raise UserError(_("Unable to change the HSN/SAC code. Please ensure all POS sessions are closed before proceeding."))
```

Comment:
```suggestion
```
Let's not do that.  (onchanges are not made for that to begin with)  
I wonder why you say it is that bad that you would correct an HSN in the middle of a PoS session?  If it was wrong and you correct it, then it will put the correct one...  @ajjo-odoo

Reply:
@zepa-odoo @jva-odoo